omod-common/src/main/java/org/openmrs/module/webservices/rest/web/filter/AuthorizationFilter.java

```diff
@@ -84,19 +85,31 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 			if (!Context.isAuthenticated()) {
 				String basicAuth = httpRequest.getHeader("Authorization");
 				if (basicAuth != null) {
-					// this is "Basic ${base64encode(username + ":" + password)}"
-					try {
-						basicAuth = basicAuth.substring(6); // remove the leading "Basic "
-						String decoded = new String(Base64.decodeBase64(basicAuth), Charset.forName("UTF-8"));
-						String[] userAndPass = decoded.split(":");
-						Context.authenticate(userAndPass[0], userAndPass[1]);
-						if (log.isDebugEnabled())
-							log.debug("authenticated " + userAndPass[0]);
-					}
-					catch (Exception ex) {
-						// This filter never stops execution. If the user failed to
-						// authenticate, that will be caught later.
+					if (basicAuth != null && basicAuth.startsWith("Basic")) {
+						// this is "Basic ${base64encode(username + ":" + password)}"
+						if (StringUtils.isBlank(basicAuth) || !basicAuth.contains(":")) {
```

Comment:
Yeah, this block needs to be after we create `decoded` and then:
```suggestion
						if (StringUtils.isBlank(decoded) || !decoded.contains(":")) {
```